terraform/functions.tf

```diff
@@ -683,5 +694,150 @@ module "lambda_function-treasuryReportGeneration" {
       principal  = "s3.amazonaws.com"
       source_arn = module.reporting_data_bucket.bucket_arn
     }
+    StepFunctionTrigger = {
+      principal  = "states.amazonaws.com"
+      source_arn = module.step_function.arn
+    }
   }
 }
+
+
+module "step_function" {
+  source = "terraform-aws-modules/step-functions/aws"
+
+  name       = "generate-treasury-report"
+  definition = <<EOF
+{
```

Comment:
Do not use this since this causes issues with the linter locally.

Provide a HCL terraform object and wrap it in jsonencode(``)

Reply:
another option is to define the definition as a terraform local and pass it in as a one-liner. and use the local to define as an output.